src/generators/operationGenerator.ts

```diff
@@ -151,45 +153,187 @@ function buildSpec(spec: OperationSpecDetails): string {
   const contentType = buildContentType(spec);
   const mediaType = buildMediaType(spec);
 
-  const isXML = spec.isXML ? "isXML: true," : "";
   const serializerName = spec.isXML
     ? "serializer: xmlSerializer"
     : "serializer";
 
-  return `{ path: "${spec.path}", httpMethod: "${
-    spec.httpMethod
-  }", responses: {${responses.join(
-    ", "
-  )}},${requestBody}${formDataParams}${queryParams}${urlParams}${headerParams}${isXML}${contentType}${mediaType}${serializerName}
-    }`;
+  writer.block(() => {
+    writer.write(`path: "${spec.path}",`);
+    writer.write(`httpMethod: "${spec.httpMethod}",`);
+    writer.write(`responses: { ${responses.join(", ")} },`);
+
+    if (typeof requestBody === "string") {
+      writer.write(requestBody);
+    } else {
+      const mapper =
+        typeof requestBody.mapper === "string"
+          ? `Mappers.${requestBody.mapper}`
+          : `${JSON.stringify(requestBody.mapper)}`;
+
+      writer.write(
+        `requestBody: { parameterPath: ${JSON.stringify(
+          requestBody.parameterPath
+        )}, mapper: ${mapper}}, `
+      );
+    }
+
+    if (formDataParams) {
+      writer.write(formDataParams);
+      writer.write(", ");
```

Comment:
Also decided to explicitly add the commas here rather than within the helper `build*` functions to make it easier to manage